dds_web/crypt/auth.py

```diff
@@ -2,10 +2,86 @@
 
 import argon2
 
-from sqlalchemy.exc import SQLAlchemyError
+import sqlalchemy
 
 from dds_web.database import models
 from sqlalchemy.sql import func
+from dds_web import exceptions
+from flask import session
+
+
+def verify_user_pass(username, password):
+    """Verify that user exists and password is correct."""
+
+    # Verify existing user
+    try:
+        user = models.User.query.filter(models.User.username == func.binary(username)).first()
+    except sqlalchemy.exc.SQLAlchemyError:
+        raise
+
+    # User does not exist
+    if not user:
+        raise exceptions.AuthenticationError("Incorrect username and/or password!")
+
+    # Verify password and generate token
+    if verify_password_argon2id(user.password, password):
+        # TODO (ina): Look into moving this (below)
+        # if "l" not in list(user.permissions):
+        #     raise exceptions.AuthenticationError(
+        #         f"The user '{username}' does not have any permissions"
+        #     )
+
+        # Password correct
+        return True
+
+    # Password incorrect
+    return False
+
+
+def user_session_info(username):
```

Comment:
Why do we need to load the following information in the session? I think the username should already be kept in the authenticated session. The other information such as is_admin and is_facility can be checked on a need to check basis. The way we keep it in session causes extra work and can cause inconsistency issues as the data needs to be updated in several places such as both db and session. I recommend keeping the session variables to a minimum and only when really needed.

Reply:
Fair enough. Basically just created this function from another old function. But this is only used in the web interface - should I leave it for now since we're focusing on the REST API and CLI ?